File: magellan-sample-migration/src/main/java/com/wealthfront/magellan/sample/migration/AppComponent.java
package com.wealthfront.magellan.sample.migration;

import com.wealthfront.magellan.sample.migration.tide.DogDetailsScreen;
import com.wealthfront.magellan.sample.migration.tide.DogListStep;

import javax.inject.Singleton;

import dagger.Component;

@Component(modules = AppModule.class)
@Singleton
public interface AppComponent {

  void inject(MainActivity activity);

  void inject(DogListStep step);

  void inject(DogDetailsScreen screen);

  void inject(Expedition expedition);
}

File: magellan-sample-migration/src/main/java/com/wealthfront/magellan/sample/migration/AppComponent.kt
package com.wealthfront.magellan.sample.migration

import dagger.Component
import javax.inject.Singleton

@Component(modules = [AppModule::class, DogApiModule::class, ToolbarHelperModule::class])
@Singleton
interface AppComponent {

  fun inject(activity: MainActivity)
  fun inject(expedition: Expedition)
}
